internal/config/config_test.go

```diff
@@ -0,0 +1,156 @@
+package config
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+var requiredConfVars = []string{
+	"db.host", "db.user", "db.password", "db.database", "c4gh.filepath", "c4gh.passphrase", "oidc.ConfigurationURL",
+}
+
+type TestSuite struct {
+	suite.Suite
+}
+
+func (suite *TestSuite) SetupTest() {
+	viper.Set("db.host", "test")
+	viper.Set("db.user", "test")
+	viper.Set("db.password", "test")
+	viper.Set("db.database", "test")
+	viper.Set("c4gh.filepath", "test")
+	viper.Set("c4gh.passphrase", "test")
+	viper.Set("oidc.ConfigurationURL", "test")
+}
+
+func (suite *TestSuite) TearDownTest() {
+	viper.Reset()
+}
+
+func TestConfigTestSuite(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+func (suite *TestSuite) TestConfigFile() {
+	viper.Set("configFile", "test")
+	config, err := NewConfig()
+	assert.Nil(suite.T(), config)
+	assert.Error(suite.T(), err)
+	assert.Equal(suite.T(), "test", viper.ConfigFileUsed())
+}
+
+func (suite *TestSuite) TestMissingRequiredConfVar() {
+	for _, requiredConfVar := range requiredConfVars {
+		requiredConfVarValue := viper.Get(requiredConfVar)
+		viper.Set(requiredConfVar, nil)
+		expectedError := fmt.Errorf("%s not set", requiredConfVar)
+		config, err := NewConfig()
+		assert.Nil(suite.T(), config)
+		if assert.Error(suite.T(), err) {
+			assert.Equal(suite.T(), expectedError, err)
+		}
+		viper.Set(requiredConfVar, requiredConfVarValue)
+	}
+}
+
+func (suite *TestSuite) TestAppConfig() {
+	// Test fail on key read error
+	viper.Set("app.host", "test")
+	viper.Set("app.port", 1234)
+	viper.Set("app.tlscert", "test")
+	viper.Set("app.tlskey", "test")
+	viper.Set("app.archivePath", "/test")
+	viper.Set("app.logLevel", "debug")
+
+	viper.Set("db.sslmode", "disable")
+
+	_, err := NewConfig()
+	assert.Error(suite.T(), err, "Error expected")
+
+	// Test pass on key read
+	originalGetC4GHKey := GetC4GHKey
+	GetC4GHKey = func() (*[32]byte, error) {
+		return nil, nil
+	}
```

Comment:
```suggestion
	_, privateKey, err := keys.GenerateKeyPair()
	assert.NoError(suite.T(), err)
	tempDir := suite.T().TempDir()
	privateKeyFile, err := os.Create(fmt.Sprintf("%s/c4fg.key", tempDir))
	err = keys.WriteCrypt4GHX25519PrivateKey(privateKeyFile, privateKey, []byte("password"))
	viper.Set("c4gh.filepath", fmt.Sprintf("%s/c4fg.key", tempDir))
	viper.Set("c4gh.passphrase", "password")
```

This is one way of generating the c4gh key during testing

Reply:
Tested this, and it works nicely